promotion/instrumented_datastore.go

```diff
@@ -430,6 +416,20 @@ func (_d DatastoreWithPrometheus) RollbackTx(tx *sqlx.Tx) {
 	return
 }
 
+// RollbackTxAndHandle implements Datastore
+func (_d DatastoreWithPrometheus) RollbackTxAndHandle(tx *sqlx.Tx) (err error) {
+	_since := time.Now()
+	defer func() {
+		result := "ok"
+		if err != nil {
```

Comment:
line 430 we are returning the base of RollbackTxAndHandle.  Since we defined the `err` var on 420 as the response basically go instanciates this value at the moment the functino RollbackTxAndHandle is created.  Since we are in a `defer`-ed function call we can reference the return value `err` in this case.  Here is the callstack:

DatastoreWithPrometheus.RollbackTxAndHandle
  - var tx passed by reference
  - err instanciated and placed on stack
DatastoreWithPrometheus.base.RollbackTxAndHandle called
DatastoreWithPrometheus.base.RollbackTxAndHandle result (an error) is assigned to err which is on the stack
defer func() is called
  - err is the variable from a parent scope from where it was created (the error which is on the stack)

I agree it is hard to read, but basically this is checking if base returned an error, and this is all happening after RollbackTxAndHandle completes in the deferred anonymous function.